lib/agentless/agentless.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agentless
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/utils/sshutils"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/native"
+)
+
+// SiteClientGetter returns an auth client to a given cluster.
+type SiteClientGetter interface {
+	GetSiteClient(ctx context.Context, clusterName string) (auth.ClientI, error)
+}
+
+// CreateAuthSigner attempts to create a [ssh.Signer] that is signed with
+// OpenSSH CA and can be used to authenticate to agentless nodes.
+func CreateAuthSigner(ctx context.Context, username, clusterName string, ttl time.Duration, clientGetter SiteClientGetter) (ssh.Signer, error) {
+	// generate a new key pair
+	priv, err := native.GeneratePrivateKey()
```

Comment:
This always generates RSA keys, and since we know in this case we're connecting to a OpenSSH server, and OpenSSH has supported ed25519 keys since 6.5, it would be better to generate ed25519 keys. Maybe in another PR I could extend the `native` package to optionally generate ed25519 keys and use that here?

Reply:
@capnspacehook Let's stick to RSA. OpenSSH had a few bugs related to ed25519 that I'm aware of and probably a few more that I'm not. We know that RSA works with our implementation. We can add support for it at some point, but for now, let's stick to the evil that we know.